### ISSUE TITLE
Syntax error of the upstream master

### ISSUE THREAD

**Issue by tobegit3hub** [astaxie/beego#1696]
When I try to `go get` the latest beego and get errors.

```
➜  go get -u github.com/astaxie/beego
# github.com/astaxie/beego
../../astaxie/beego/tree.go:144: syntax error: unexpected range, expecting {
../../astaxie/beego/tree.go:155: syntax error: unexpected else, expecting semicolon or newline
../../astaxie/beego/tree.go:157: non-declaration statement outside function body
../../astaxie/beego/tree.go:158: non-declaration statement outside function body
../../astaxie/beego/tree.go:159: non-declaration statement outside function body
../../astaxie/beego/tree.go:160: syntax error: unexpected }
../../astaxie/beego/tree.go:257: syntax error: unexpected range, expecting {
../../astaxie/beego/tree.go:267: syntax error: unexpected else, expecting semicolon or newline or }
../../astaxie/beego/tree.go:283: syntax error: unexpected }
```

And the same error occurs when `go build` to beego project.

```
➜  beego git:(master) git pull
Already up-to-date.
➜  beego git:(master) go build
# github.com/astaxie/beego
./tree.go:144: syntax error: unexpected range, expecting {
./tree.go:155: syntax error: unexpected else, expecting semicolon or newline
./tree.go:157: non-declaration statement outside function body
./tree.go:158: non-declaration statement outside function body
./tree.go:159: non-declaration statement outside function body
./tree.go:160: syntax error: unexpected }
./tree.go:257: syntax error: unexpected range, expecting {
./tree.go:267: syntax error: unexpected else, expecting semicolon or newline or }
./tree.go:283: syntax error: unexpected }
```

**Comment by ysqi** [comment]
hi @tobegit3hub ,What's your Go version ?  this bug fixed in dev branch ,see #1623 .

**Comment by tobegit3hub** [comment]
Thanks @ysqi and it's the same problem of https://github.com/astaxie/beego/issues/1623 .

I'm gonna close this because it works well in golang 1.4+ .
